### DISCUSSION TITLE
Customize data in relationmanger prior saving in DB

### DISCUSSION THREAD

**Discussion by towerthepower89** [filamentphp/filament#12746] [Help]
### Package

Form builder

### Package Version

v3.0.0.0

### How can we help you?

Hi,

I have a question and after many searches, I was not able to find the answer to my problem.

Let's suppose I have 2 classes, Work(Job) and User .

Work can be completed by one or multiple user, in percentage. Each Work costs a certain amount of money.

I create the work, and while saving the data in the db, the user who added the work is also attached to the same, via relationmanager. (There is a second table with fields work_id, user_id, percentage, expected_salary). For achieving this, I add a new record in the second table using handleRecordCreation, in the CreateWork page, under App\Filament\Resources\WorkResource\Pages.

`
protected function handleRecordCreation(array $data): Model
{
	//get the inserted work
	$record =  static::getModel()::create($data);

	// Create a new worker for the model instance
	$workuser = new Workuser();
	$workuser->work_id = $record->id;
	$workuser->user_id = auth()->user()->id;
	$workuser->percentage = '100';
	$workuser->expected_salary = $record->totalPayment; //assuming only one user for this job
	$workuser->save();

	return $record;
}`

All this works great :) 

Next step is to add more users to the same work. So I go to Work page, click edit on the row I need, and on the bottom of the Edit page there are the relationships. I click on New, and a new user is connected to this work. 
I would like to:

-  directly calculate the expected_salary based on the percentage of work done by this user. I was thinking to calculate it while inserting, but I am unable to get  the (parent) work->cost while inserting a new record, in the form.
- So I was thinking to customize the saving query, and get that information while saving. But also in this case, I got stuck, since there is no CreateRecord page for the resourcemanager, and if I add handleRecordCreation in the Workuser page, it doesn't get called.

Do anybody know how to customize the saving query, or the data, before inserting a new resourcemanager record?

Thank you!

**Comment by micraux** [comment]
Hi,

I don't know if I understood well your question, but maybe you could use a Repeater inside the Work create form?
